mlir/lib/Dialect/SPIRV/Transforms/UpdateVCEPass.cpp

```diff
@@ -95,6 +95,18 @@ static LogicalResult checkAndUpdateCapabilityRequirements(
   return success();
 }
 
+static void addAllImpliedCapabilities(SetVector<spirv::Capability> &caps) {
+  size_t old_size{0};
+  while (caps.size() > old_size) {
+    old_size = caps.size();
+    SetVector<spirv::Capability> tmp;
+    for (spirv::Capability cap : caps)
+      tmp.insert_range(getDirectImpliedCapabilities(cap));
```

Comment:
Sorry I just figured out that there is a `getRecursiveImpliedCapabilities` method.
I think this can be simplified further.